packages/domain-lib/test/unit/index.test.ts

```diff
@@ -1053,4 +1053,143 @@ describe("Settlements BC [Domain] - Unit Tests", () => {
+		expect(batchDisp!.state).toEqual('DISPUTED');
+
+		//check matrix2 before reclaculate
+		matrixDisp2 = await aggregate.getSettlementMatrix(securityContext, matrixIdDisp2);
+		expect(matrixDisp2).toBeDefined();
+		expect(matrixDisp2!.id).toEqual(matrixIdDisp2);
+		expect(matrixDisp2!.state).toEqual('DISPUTED');
+		expect(matrixDisp2!.type).toEqual('STATIC');
+		expect(matrixDisp2!.participantBalances.length).toEqual(2);
+		expect(matrixDisp2!.totalDebitBalance).toEqual("10000");
+		expect(matrixDisp2!.totalCreditBalance).toEqual("10000");
+		expect(matrixDisp2!.participantBalancesDisputed.length).toEqual(0);
+		expect(matrixDisp2!.totalDebitBalanceDisputed).toEqual("0");
+		expect(matrixDisp2!.totalCreditBalanceDisputed).toEqual("0");
+		
+		await aggregate.recalculateSettlementMatrix(securityContext, matrixIdDisp2);
+
+		//check matrix2 after reclaculate
+		matrixDisp2 = await aggregate.getSettlementMatrix(securityContext, matrixIdDisp2);
+		expect(matrixDisp2).toBeDefined();
+		expect(matrixDisp2!.id).toEqual(matrixIdDisp2);
+		expect(matrixDisp2!.state).toEqual('DISPUTED');
+		expect(matrixDisp2!.type).toEqual('STATIC');
+		expect(matrixDisp2!.participantBalances.length).toEqual(0);
+		expect(matrixDisp2!.totalDebitBalance).toEqual("0");
+		expect(matrixDisp2!.totalCreditBalance).toEqual("0");
+		expect(matrixDisp2!.participantBalancesDisputed.length).toEqual(2);
+		expect(matrixDisp2!.totalDebitBalanceDisputed).toEqual("10000");
+		expect(matrixDisp2!.totalCreditBalanceDisputed).toEqual("10000");
+
+		//check matrix1 again
+		await aggregate.recalculateSettlementMatrix(securityContext, matrixIdDisp);
+
+		matrixDisp = await aggregate.getSettlementMatrix(securityContext, matrixIdDisp);
+
+		//check before settlement
+		expect(matrixDisp).toBeDefined();
+		expect(matrixDisp!.id).toEqual(matrixIdDisp);
+		expect(matrixDisp!.state).toEqual('IDLE');
+		expect(matrixDisp!.type).toEqual('STATIC');
+		expect(matrixDisp!.participantBalances.length).toEqual(2);
+		expect(matrixDisp!.totalDebitBalance).toEqual("10000");
+		expect(matrixDisp!.totalCreditBalance).toEqual("10000");
+		expect(matrixDisp!.participantBalancesDisputed.length).toEqual(2);
+		expect(matrixDisp!.totalDebitBalanceDisputed).toEqual("10000");
+		expect(matrixDisp!.totalCreditBalanceDisputed).toEqual("10000");
+		
+		//settle matrix1
+		await aggregate.settleSettlementMatrix(securityContext, matrixIdDisp);
+		
+		matrixDisp = await aggregate.getSettlementMatrix(securityContext, matrixIdDisp);
+
+		//check the settled matrix1 again
+		expect(matrixDisp).toBeDefined();
+		expect(matrixDisp!.id).toEqual(matrixIdDisp);
+		expect(matrixDisp!.state).toEqual('SETTLED');
+		expect(matrixDisp!.type).toEqual('STATIC');
+		expect(matrixDisp!.participantBalances.length).toEqual(2);
+		expect(matrixDisp!.totalDebitBalance).toEqual("10000");
+		expect(matrixDisp!.totalCreditBalance).toEqual("10000");
+		expect(matrixDisp!.participantBalancesDisputed.length).toEqual(2);
```

Comment:
@koekiebox , my test case will be failed here because matrix does not have disputed batch info after settled.